### PR TITLE
`PodLifeTime`:  update support pods with container status and  pods reason

### PR DESCRIPTION
**Pull request by dongjiang1989**
support pod status reason:
- NodeAffinity
- NodeLost
- Shutdown
- UnexpectedAdmissionError

support pod with container status:
- CrashLoopBackOff
- CreateContainerConfigError
- ErrImagePull
- CreateContainerError
- InvalidImageName

### CODE REVIEW

File: pkg/framework/plugins/podlifetime/validation.go [hunk 1]
```diff
@@ -44,15 +44,30 @@ func ValidatePodLifeTimeArgs(obj runtime.Object) error {
...
+		"NodeLost",
+		"Shutdown",
+		"UnexpectedAdmissionError",
+
+		// Container Status Reasons
 		// Container state reasons: https://github.com/kubernetes/kubernetes/blob/release-1.24/pkg/kubelet/kubelet_pods.go#L76-L79
 		"PodInitializing",
 		"ContainerCreating",
 
 		// containerStatuses[*].state.waiting.reason: ImagePullBackOff
```
Comment (dongjiang1989) [inline pkg/framework/plugins/podlifetime/validation.go:63]:
```suggestion
		// containerStatuses[*].state.waiting.reason: ImagePullBackOff, etc.
```

File: pkg/framework/plugins/podlifetime/validation.go [hunk 2]
```diff
@@ -44,15 +44,30 @@ func ValidatePodLifeTimeArgs(obj runtime.Object) error {
...
 		"PodInitializing",
 		"ContainerCreating",
 
-		// containerStatuses[*].state.waiting.reason: ImagePullBackOff
+		// containerStatuses[*].state.waiting.reason: ImagePullBackOff, etc.
+		"ImagePullBackOff",
+		"CrashLoopBackOff",
+		"CreateContainerConfigError",
+		"ErrImagePull",
 		"ImagePullBackOff",
```
Comment (dongjiang1989) [inline pkg/framework/plugins/podlifetime/validation.go:68]:
```suggestion
```

**Review by dongjiang1989 — COMMENTED** [review]
Remove duplicates

**Comment by dongjiang1989** [comment]
@a7i  PTAL

File: pkg/framework/plugins/podlifetime/pod_lifetime_test.go [hunk 3]
```diff
@@ -337,6 +337,38 @@ func TestPodLifeTime(t *testing.T) {
 				}
 			},
 		},
+		{
```
Comment (jklaw90) [inline pkg/framework/plugins/podlifetime/pod_lifetime_test.go:340]: can you add tests for each status that you are adding

Comment (dongjiang1989) [reply pkg/framework/plugins/podlifetime/pod_lifetime_test.go:346]: Thanks for your review. 
Done. 
Please re-check it

File: pkg/framework/plugins/podlifetime/validation.go [hunk 4]
```diff
@@ -44,15 +44,29 @@ func ValidatePodLifeTimeArgs(obj runtime.Object) error {
...
+		// Pod Status Phase
 		string(v1.PodRunning),
 		string(v1.PodPending),
+		string(v1.PodUnknown),
 
+		// Pod Status Reasons
+		"NodeAffinity",
+		"NodeLost",
+		"Shutdown",
+		"UnexpectedAdmissionError",
```
Comment (a7i) [inline pkg/framework/plugins/podlifetime/validation.go:56]: Some (maybe all) of these are status reasons for Pods that have status Failed, which generally has been handled by `RemoveFailedPods` strategy as opposed to `PodLifeTime`.

We have discussed merging the two strategies in the past so I'm not opposed to this idea.

Let's give it a couple of more days to see if other maintainers want to chime in on our approach going forward.

Comment (dongjiang1989) [reply pkg/framework/plugins/podlifetime/validation.go:56]: Great. I’m willing to join and improve it

**Comment by dongjiang1989** [comment]
> /label tide/merge-method-squash

squash done. @a7i PTAL, thanks

**Pull request merged**